engines/router/missionctl/mission_control_upi.go

```diff
@@ -0,0 +1,97 @@
+package missionctl
+
+import (
+	"context"
+
+	"github.com/caraml-dev/turing/engines/router/missionctl/errors"
+	"github.com/caraml-dev/turing/engines/router/missionctl/fiberapi"
+	"github.com/caraml-dev/turing/engines/router/missionctl/instrumentation/metrics"
+	upiv1 "github.com/caraml-dev/universal-prediction-interface/gen/go/grpc/caraml/upi/v1"
+	"github.com/gojek/fiber"
+	fibergrpc "github.com/gojek/fiber/grpc"
+	"google.golang.org/grpc"
+	"google.golang.org/protobuf/proto"
+)
+
+type MissionControlUPI interface {
+	Route(context.Context, fiber.Request) (*upiv1.PredictValuesResponse, *errors.TuringError)
+}
+
+type missionControlUpi struct {
+	fiberRouter fiber.Component
+}
+
+// NewMissionControlUPI creates new instance of the MissingControl,
+// based on the grpc configuration of fiber.yaml
+func NewMissionControlUPI(
+	cfgFilePath string,
+	fiberDebugLog bool,
+) (MissionControlUPI, error) {
+	fiberRouter, err := fiberapi.CreateFiberRouterFromConfig(cfgFilePath, fiberDebugLog)
+	if err != nil {
+		return nil, err
+	}
+
+	return &missionControlUpi{
+		fiberRouter: fiberRouter,
+	}, nil
+}
+
+func (us *missionControlUpi) Route(
+	ctx context.Context,
+	fiberRequest fiber.Request) (
+	*upiv1.PredictValuesResponse, *errors.TuringError) {
```

Comment:
What happen if we return dynamic message here?

Reply:
Asking to understand whether it can be used to address https://github.com/caraml-dev/turing/pull/238#discussion_r969130672

Reply:
we can but we would still need to do the marshaling in the UPIServer to satisfy the interface regardless? 

```
PredictValues(ctx context.Context, req *upiv1.PredictValuesRequest) (*upiv1.PredictValuesResponse, error) 
```

Reply:
Ah yes, you are right